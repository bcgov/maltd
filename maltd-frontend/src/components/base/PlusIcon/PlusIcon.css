.inline-blocked {
  display: inline-block !important;
  margin-top: -10px;
  margin-left: 20px;
}
