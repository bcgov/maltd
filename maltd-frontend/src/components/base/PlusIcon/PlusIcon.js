import React from "react";
import PropTypes from "prop-types";
import Icon from "../../../img/plus-icon.PNG";
import "./PlusIcon.css";

export default function PlusIcon({ onClick }) {
  return (
    <div
      role="button"
      tabIndex="0"
      onClick={onClick}
      onKeyDown={onClick}
      className="icon"
    >
      <img src={Icon} alt="plus icon" width="25px" height="25px" />
    </div>
  );
}

PlusIcon.propTypes = {
  onClick: PropTypes.func.isRequired
};
