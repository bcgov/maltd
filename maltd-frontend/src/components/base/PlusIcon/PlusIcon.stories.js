import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import "bootstrap/dist/css/bootstrap.css";
import PlusIcon from "./PlusIcon";

storiesOf("PlusIcon", module).add("default", () => (
  <PlusIcon onClick={action("plus icon clicked")} />
));
