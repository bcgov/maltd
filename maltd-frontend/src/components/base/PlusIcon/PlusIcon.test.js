import React from "react";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { shallow } from "enzyme";
import PlusIcon from "./PlusIcon";

Enzyme.configure({ adapter: new Adapter() });

describe("Plus Icon", () => {
  test("Component renders as expected", () => {
    const component = renderer.create(<PlusIcon onClick={() => jest.fn()} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Component onClick and onKeyDown work as expected", () => {
    const testState = { isClicked: false };
    const onClick = () => {
      testState.isClicked = true;
    };

    const wrapper = shallow(
      <PlusIcon id="123" onKeyDown={onClick} onClick={onClick} />
    );

    wrapper.find("#plus-icon").simulate("keydown");
    expect(testState.isClicked).toEqual(true);

    wrapper.find("#plus-icon").simulate("click");
    expect(testState.isClicked).toEqual(true);
  });
});
