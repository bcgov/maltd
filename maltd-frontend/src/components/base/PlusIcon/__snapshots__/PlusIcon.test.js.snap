// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plus Icon Component renders as expected 1`] = `
<div
  className="inline-blocked"
  id="main"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <img
    alt="plus icon"
    height="25px"
    src="plus-icon.PNG"
    width="25px"
  />
</div>
`;
